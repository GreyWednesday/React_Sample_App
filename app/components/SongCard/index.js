import React, { memo } from 'react';
import { Col } from 'antd';

import styled from 'styled-components';
import PropTypes from 'prop-types';
import * as colors from '@app/themes/colors';
import { styles } from '@app/themes/index';

const Card = styled.div`
  && {
    display: grid;
    margin: 1.25em auto;
    background: ${colors.primary};
    ${styles.borderRadius('10px')};
    grid-template-columns: repeat(2, 1fr);
    transition: ease-in-out 0.3s;
    grid-gap: 10px;
    word-wrap: break-word
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    justify-items: center;
  }

  :hover {
    ${styles.borderRadius('20px')};
    transform: scale(1.1);
    ${styles.boxShadow(5, 5, 30, 0)};
  }

  audio {
    grid-column: span 2;
  }
`;

const Artwork = styled.img`
  border-radius: 10px;
  margin: 10px auto;
`;

const Title = styled.h3`
  margin: auto;
`;

const Collection = styled.h5`
  margin: auto;
`;

function SongCard({ song }) {
  return (
    <Col span={8} data-testid="song-card">
      <a href={`/tracks/${song ? song.trackId : ''}/`} data-testid="card-link">
        <Card>
          <Artwork src={song ? song.artworkUrl100 : ''} />
          <Title>{song ? song.trackName : ''}</Title>
          <Title>Collection Name : </Title>
          <Collection>
            {song ? `${song.collectionName ? song.collectionName : 'No Information available'}` : ''}
          </Collection>
          <audio controls>
            <source src={song ? song.previewUrl : ''} type="audio/mpeg" />
          </audio>
        </Card>
      </a>
    </Col>
  );
}

SongCard.propTypes = {
  song: PropTypes.object
};

export default memo(SongCard);
