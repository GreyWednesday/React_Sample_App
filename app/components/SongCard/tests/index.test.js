/**
 *
 * Tests for SongCard
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl, renderProvider } from '@utils/testUtils';
import SongCard from '../index';

describe('<SongCard />', () => {
  let song;

  beforeEach(() => {
    song = {
      trackId: 1000,
      artworkUrl100: 'react-template.png',
      trackName: 'abc',
      previewUrl:
        'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/5e/5b/3d/5e5b3df4-deb5-da78-5d64-fe51d8404d5c/mzaf_13341178261601361485.plus.aac.p.m4a'
    };
  });

  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<SongCard song={song} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 SongCard component', () => {
    const { getAllByTestId } = renderWithIntl(<SongCard song={song} />);
    expect(getAllByTestId('song-card').length).toBe(1);
  });

  it('should render even with no image, audio or trackId or trackName', () => {
    const { baseElement } = renderWithIntl(<SongCard song={song} />);
    expect(baseElement).toBeTruthy();
  });

  it('redirects to the correct link', () => {
    const { getByTestId } = renderProvider(<SongCard song={song} />);
    expect(getByTestId('card-link')).toHaveAttribute('href', '/tracks/1000/');
  });
});
