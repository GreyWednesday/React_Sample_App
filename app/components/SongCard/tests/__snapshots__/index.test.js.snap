// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SongCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-col ant-col-8"
      data-testid="song-card"
    >
      <a
        data-testid="card-link"
        href="/tracks/1000/"
      >
        <div
          class="SongCard__Card-sc-163fh4j-0 kqBnJO"
        >
          <img
            class="SongCard__Artwork-sc-163fh4j-1 jDqjW"
            src="react-template.png"
          />
          <h3
            class="SongCard__Title-sc-163fh4j-2 iDAhMX"
          >
            abc
          </h3>
          <h3
            class="SongCard__Title-sc-163fh4j-2 iDAhMX"
          >
            Collection Name : 
          </h3>
          <h5
            class="SongCard__Collection-sc-163fh4j-3 ejeFjk"
          >
            No Information available
          </h5>
          <audio
            controls=""
          >
            <source
              src="https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/5e/5b/3d/5e5b3df4-deb5-da78-5d64-fe51d8404d5c/mzaf_13341178261601361485.plus.aac.p.m4a"
              type="audio/mpeg"
            />
          </audio>
        </div>
      </a>
    </div>
  </div>
</body>
`;
